### PR TITLE
introduce labels for resources to be backed up

### PR DESCRIPTION
**Pull request by wkloucek**
## Description
Introduces a new option to set labels on resources that must be backed up.

## Related Issue
- Compare to https://doc.owncloud.com/ocis/5.0/maintenance/b-r/backup_considerations.html
- see also OCSRE-92

## Motivation and Context
Currently there is no other way to inject labels to specific resources

## How Has This Been Tested?
- installing the develoment example deployment with following changes:

```diff
diff --git a/deployments/development-install/helmfile.yaml b/deployments/development-install/helmfile.yaml
index 6edf79a..0bdd3b3 100644
--- a/deployments/development-install/helmfile.yaml
+++ b/deployments/development-install/helmfile.yaml
@@ -14,6 +14,17 @@ releases:
               hosts:
                 - ocis.kube.owncloud.test
 
+      - backup:
+          configMapLabels:
+            backup.test/include: "true"
+            backup.test/configmap: "test"
+          secretLabels:
+            backup.test/include: "true"
+            backup.test/secret: "test"
+          pvcLabels:
+            backup.test/include: "true"
+            backup.test/pvc: "test"
+
       - logging:
           level: debug
```

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation generated (`make docs`) and committed
- [ ] Documentation ticket raised: <link>
- [ ] Documentation PR created: <link>

### CODE REVIEW

File: charts/ocis/templates/authmachine/secret.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{- if or (not .Values.secretRefs.machineAuthApiKeySecretRef) }}
 {{- $params := (dict)}}
 {{- $_ := set $params "machine-auth-api-key" (randAlphaNum 30) }}
-{{- include "ocis.secret" (dict "scope" . "name" "machine-auth-api-key" "params" $params)}}
+{{- include "ocis.secret" (dict "scope" . "name" (include "secrets.machineAuthAPIKeySecret" .) "params" $params)}}
```
Comment (d7oc) [inline charts/ocis/templates/authmachine/secret.yaml:4]: Where does this come from. I don't see that a new template for `"secrets.machineAuthAPIKeySecret"` was added here.

Comment (wkloucek) [reply charts/ocis/templates/authmachine/secret.yaml:4]: It's not new, it only was used in a single place before. This change removes duplicated declaration of the constant (the reason why we might have introduced the template in the first place)

**Review by d7oc — COMMENTED** [review]
The following PVC were not added:

```
./charts/ocis/templates/web/pvc.yaml
./charts/ocis/templates/search/pvc.yaml
./charts/ocis/templates/nats/pvc.yaml
./charts/ocis/templates/thumbnails/pvc.yaml
```

Was this by intention?

What was the basis for decision about which secrets are backed up and which not? E.g why are 

"secrets.jwtSecret"
"secrets.adminUser"

Not backed up?

Similar question for the config map. I'm missing information what e.g. the Graph application id isn't relevant for a backup.

**Comment by wkloucek** [comment]
@d7oc this should answer your questions (see also the description)

> * Compare to https://doc.owncloud.com/ocis/5.0/maintenance/b-r/backup_considerations.html


> Could we please split cleanups from the basic topic of the PRs. 

I can do it

**Review by d7oc — APPROVED** [review]
LGTM

**Pull request merged**